### PR TITLE
Add display columns for appcatalogentry CRD

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#10930

The display columns plus the latest label gives a nice overview of the current apps.

```bash
$ kubectl get appcatalogentry -l latest=true

NAME                                                     CATALOG                 APP NAME                       APP VERSION    VERSION   AGE
giantswarm-aqua-app-enforcer-4.6.3                       giantswarm              aqua-app-enforcer              4.6.0          4.6.3     50d
giantswarm-aqua-app-scanner-4.6.3                        giantswarm              aqua-app-scanner               4.6.0          4.6.3     50d
giantswarm-aqua-app-server-4.6.3                         giantswarm              aqua-app-server                4.6.0          4.6.3     50d
giantswarm-azure-azure-logs-workspace-app-0.0.0          giantswarm-azure        azure-logs-workspace-app       v0.0.1         0.0.0     119d
giantswarm-cert-manager-app-2.3.0                        giantswarm              cert-manager-app               1.0.2          2.3.0     17d
giantswarm-efk-stack-app-0.3.2                           giantswarm              efk-stack-app                  1.9.0          0.3.2     37d
giantswarm-kong-app-1.0.0                                giantswarm              kong-app                       2.1            1.0.0     9d
giantswarm-nginx-ingress-controller-app-1.10.0           giantswarm              nginx-ingress-controller-app   v0.40.2        1.10.0    15d
giantswarm-playground-app-mesh-app-0.4.0                 giantswarm-playground   app-mesh-app                   0.3.0          0.4.0     245d
giantswarm-playground-azure-ad-pod-identity-app-0.2.1    giantswarm-playground   azure-ad-pod-identity-app      1.6.1          0.2.1     3d10h
giantswarm-playground-eventrouter-app-0.0.4              giantswarm-playground   eventrouter-app                v0.3           0.0.4     112d
giantswarm-playground-fluent-logshipping-app-0.5.2       giantswarm-playground   fluent-logshipping-app         0.5.2          0.5.2     85d
giantswarm-playground-gatling-app-1.0.2                  giantswarm-playground   gatling-app                    3.2.1          1.0.2     142d
giantswarm-playground-giantswarm-todo-app-0.4.1          giantswarm-playground   giantswarm-todo-app            0.4.1          0.4.1     35d
giantswarm-playground-inlets-operator-app-0.1.0          giantswarm-playground   inlets-operator-app            0.7.0          0.1.0     206d
giantswarm-playground-istio-operator-1.0.1               giantswarm-playground   istio-operator                 v1.6.0         1.0.1     97d
giantswarm-playground-jaeger-operator-app-0.2.1          giantswarm-playground   jaeger-operator-app            1.18.0         0.2.1     9d
giantswarm-playground-k8s-initiator-app-0.9.1            giantswarm-playground   k8s-initiator-app              v0.9.0         0.9.1     3d7h
giantswarm-playground-kibana-oss-app-0.0.6               giantswarm-playground   kibana-oss-app                 v0.1           0.0.6     3d5h
giantswarm-playground-linkerd2-app-0.4.1                 giantswarm-playground   linkerd2-app                   stable-2.8.1   0.4.1     9d
giantswarm-playground-loki-stack-app-0.2.1               giantswarm-playground   loki-stack-app                 v1.2.0         0.2.1     6d3h
giantswarm-playground-opencensus-collector-app-0.1.4     giantswarm-playground   opencensus-collector-app       0.1.11         0.1.4     49d
giantswarm-playground-strimzi-kafka-operator-app-0.1.0   giantswarm-playground   strimzi-kafka-operator-app     0.19.0         0.1.0     69d
giantswarm-prometheus-operator-app-0.4.0                 giantswarm              prometheus-operator-app        0.38.1         0.4.0     7d5h
```

## Checklist

- [x] Consider SIG UX feedback.
- [x] Update changelog in CHANGELOG.md.
- [ ] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

### CODE REVIEW

File: config/crd/v1beta1/application.giantswarm.io_appcatalogentries.yaml [hunk 1]
```diff
@@ -18,6 +39,7 @@ spec:
     plural: appcatalogentries
     singular: appcatalogentry
   scope: Namespaced
+  subresources: {}
```
Comment (rossf7) [inline config/crd/v1beta1/application.giantswarm.io_appcatalogentries.yaml:42]: Not sure why this was missing. But there is no status subresource so this is fine.

**Comment by rossf7** [comment]
Tested on `ghost`.

**Review by pipo02mix — APPROVED** [review]
LGTM

**Review by tomahawk28 — APPROVED** [review]
LGTM

**Pull request merged**